### PR TITLE
[DOCS] Expand put mapping API docs examples

### LINKED ISSUE #47248: [DOCS] Add practical example of using the reindex API to change mappings
**Describe the feature**:

Today, both the the Put Mapping API [1] and Reindex API [2] docs have blurbs that touch on the concept of reindexing to adjust an existing index's mappings.

Mapping API:

> Changing the mapping of an existing field could invalidate data that’s already indexed. If you need to change the mapping of a field, create a new index with the correct mappings and reindex your data into that index. If you only want to rename a field, consider adding an alias field.

Reindex API:

> Reindex does not attempt to set up the destination index. It does not copy the settings of the source index. You should set up the destination index prior to running a _reindex action, including setting up mappings, shard counts, replicas, etc.

Users might find it helpful if we went a step further to document a hands on example (e.g code snippets that can be run in Console) to demonstrate exactly how to use the reindex API to accomplish this.

[1] https://www.elastic.co/guide/en/elasticsearch/reference/current/indices-put-mapping.html#updating-field-mappings
[2] https://www.elastic.co/guide/en/elasticsearch/reference/current/docs-reindex.html

### PR DESCRIPTION
**Pull request by jrodewig**
### Changes

* Adds detailed examples for following tasks:
  * Add new properties to existing object fields
  * Add multi-fields to existing fields
  * Change supported mapping parameters for existing fields
  * Change the mapping of a field (with a reindex)
  * Rename a field
*  Updates the tagged regions reused in the 'Update the mapping of a field' section of our Mapping docs.

### Context

Previously most of these tasks were covered in a brief section that did not include example snippets.

Closes  #47248

### CODE REVIEW

**Comment by jrodewig** [comment]
Preview links:

- Put mapping API examples: http://elasticsearch_47462.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/indices-put-mapping.html#add-new-field-to-object
- Updated mapping docs section: http://elasticsearch_47462.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/mapping.html#update-mapping

File: docs/reference/indices/put-mapping.asciidoc [hunk 1]
```diff
@@ -166,35 +158,430 @@ PUT /my_index <1>
...
+
+
+[[rename-existing-field]]
+===== Rename a field
+
+// tag::rename-field[]
+If you want to rename a field,
+consider adding an <<alias, `alias`>> field to the index.
+// end::rename-field[]
+
```
Comment (debadair) [inline docs/reference/indices/put-mapping.asciidoc]: Should this say something about why you can't just rename the field? Something like:

Renaming a field would invalidate data already indexed under the old field name. 
Instead, add an alias field to create an alternate field name.

File: docs/reference/indices/put-mapping.asciidoc [hunk 2]
```diff
@@ -166,35 +158,430 @@ PUT /my_index <1>
...
+        "type": "keyword"
+      }
+    }
+  }
+}
+----
+
+While you cannot rename the `user_identifier`,
+you can add an alias field
+that creates an alternate field name.
```
Comment (debadair) [inline docs/reference/indices/put-mapping.asciidoc]: I'd incorporate this information as suggested above and remove it here.

**Review by debadair — APPROVED** [review]
One minor suggestion, but LGTM.

File: docs/reference/indices/put-mapping.asciidoc (hunk 1 above)
Comment (jrodewig) [reply docs/reference/indices/put-mapping.asciidoc]: Thanks @debadair. Added with 852a9e68aba.

File: docs/reference/indices/put-mapping.asciidoc (hunk 2 above)
Comment (jrodewig) [reply docs/reference/indices/put-mapping.asciidoc]: Removed with 852a9e68aba.

**Pull request merged**